### ISSUE TITLE
[OpenAL] SoundEffectInstance settings not applied when not playing

### ISSUE THREAD

**Issue by Jjagg** [MonoGame/MonoGame#5739]
Under OpenAL SoundEffectInstance instances only get a source assigned when they are playing (a source is the native resource for playing a sound in OpenAL). A source is reserved in the PlatformPlay method [here](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs#L115).
The platform specific methods for setting pitch, volume, etc. check if the instance has a source id and do not apply the setting if there is no source linked to the SFX instance (e.g. [PlatformSetPitch](when they are playing)). That means settings are only applied when the sound is playing or paused. An example of what intuitively should work, but does not is the following:
```
var instance = sfx.CreateInstance();
instance.Pitch = pitch;
instance.Play();
```

There's some comments about OpenAL [limiting the number of sources](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Audio/OpenALSoundController.cs#L78), but I don't think that's still relevant with OpenAL soft. My best guess is that the source limit was the reason sound effect instances did not get a source on creation. So I think we can just change that to reserve a source when creating a sound effect instance to fix this issue.

**Comment by mrhelmut** [comment]
OpenAL Soft is said to have a number of sources dependent of the memory available (virtually unlimited), but there is a limit of 256 on MacOS and Linux even with OpenAL Soft. I've already encountered this limit (and I believe I'm the one who put this constant and comment here).

I'll double check with an uncapped stress test tomorrow.

(Other platforms, including consoles, are affected by this issue, and that includes other properties like the volume and "is repeating" on media player. Source binding can be a touchy problem, so maybe we should look into a generic high level properties management that would work everywhere, like setting ptich/volume/etc after each Play.)

**Comment by Jjagg** [comment]
> I'll double check with an uncapped stress test tomorrow.

Thanks!

> so maybe we should look into a generic high level properties management that would work everywhere, like setting ptich/volume/etc after each Play

I had the same idea! Let's do it this way if OpenAL soft sources are limited on Mac/Linux.

**Comment by tomspilman** [comment]
> Other platforms, including consoles, are affected by this issue

Exactly....  there is no such thing as an unlimited number of sound resources.  We shouldn't design for that on any platform.

> maybe we should look into a generic high level properties management that would work everywhere, like setting ptich/volume/etc after each Play

We currently do this internally on most platforms, so making this part of the common sound code would be nice.  It will just be a difficult transition to make as we have to fix public and private platform implementations.
